src/test/java/uk/nhs/tis/trainee/notifications/service/HistoryServiceIntegrationTest.java

```diff
@@ -594,4 +596,92 @@ void shouldRebuildInAppMessageForTraineeWhenNotificationFound(NotificationType n
+    RecipientInfo recipientInfo = new RecipientInfo(TRAINEE_ID, EMAIL, TRAINEE_CONTACT);
+    TemplateInfo templateInfo
+        = new TemplateInfo(TEMPLATE_NAME, TEMPLATE_VERSION, TEMPLATE_VARIABLES);
+    TisReferenceInfo tisReferenceInfo = new TisReferenceInfo(TIS_REFERENCE_TYPE, TIS_REFERENCE_ID);
+    Instant olderTimestamp = Instant.now().minus(Duration.ofDays(1));
+
+    History history = new History(NOTIFICATION_ID, tisReferenceInfo, PROGRAMME_DAY_ONE,
+        recipientInfo, templateInfo, null, SENT_AT, READ_AT, PENDING, null, null, olderTimestamp);
+    service.save(history);
+
+    // When attempting to update with a newer timestamp
+    Instant newerTimestamp = Instant.now();
+    service.updateStatus(NOTIFICATION_ID.toString(), FAILED, "Update with newer timestamp",
+        newerTimestamp);
+
+    // Then the status should be updated
+    Optional<History> updatedHistory = service.findAllHistoryForTrainee(TRAINEE_ID).stream()
+        .filter(h -> h.id().equals(NOTIFICATION_ID))
+        .findFirst();
+    assertThat("History should be present", updatedHistory.isPresent(), is(true));
+    assertThat("Unexpected status", updatedHistory.get().status(), is(FAILED));
+    assertThat("Unexpected status detail", updatedHistory.get().statusDetail(),
+        is("Update with newer timestamp"));
+    assertThat("Unexpected status timestamp", updatedHistory.get().latestStatusEventAt(),
+        is(newerTimestamp.truncatedTo(ChronoUnit.MILLIS)));
+  }
+
+  @Test
+  void shouldNotUpdateStatusWhenNewStatusTimestampIsOlder() {
+    // Given a history record with a newer status timestamp
+    RecipientInfo recipientInfo = new RecipientInfo(TRAINEE_ID, EMAIL, TRAINEE_CONTACT);
+    TemplateInfo templateInfo
+        = new TemplateInfo(TEMPLATE_NAME, TEMPLATE_VERSION, TEMPLATE_VARIABLES);
+    TisReferenceInfo tisReferenceInfo = new TisReferenceInfo(TIS_REFERENCE_TYPE, TIS_REFERENCE_ID);
+    Instant newerTimestamp = Instant.now();
+
+    History history = new History(NOTIFICATION_ID, tisReferenceInfo, FORM_UPDATED, recipientInfo,
+        templateInfo, null, SENT_AT, READ_AT, SENT, null, null, newerTimestamp);
+    service.save(history);
+
+    // When attempting to update with an older timestamp
+    Instant olderTimestamp = Instant.now().minus(Duration.ofDays(1));
+    service.updateStatus(NOTIFICATION_ID.toString(), FAILED, "Update with older timestamp",
+        olderTimestamp);
+
+    // Then the status should not be updated
+    Optional<History> updatedHistory = service.findAllHistoryForTrainee(TRAINEE_ID).stream()
+        .filter(h -> h.id().equals(NOTIFICATION_ID))
+        .findFirst();
+    assertThat("History should be present", updatedHistory.isPresent(), is(true));
+    assertThat("Status should not change", updatedHistory.get().status(), is(SENT));
+    assertThat("Status timestamp should not change",
+        updatedHistory.get().latestStatusEventAt(),
+        is(newerTimestamp.truncatedTo(ChronoUnit.MILLIS)));
```

Comment:
Not requesting you change this, since it's consistent with previous approaches. But FYI I've started using [isCloseTo](https://hamcrest.org/JavaHamcrest/javadoc/1.3/org/hamcrest/number/IsCloseTo.html) as an alternative elsewhere for something a little more flexible/consistent.

```java
assertThat("Unexpected timestamp.", (double) submitted.getEpochSecond(), closeTo(now.getEpochSecond(), 1));
```